python-code-1.py

```diff
@@ -0,0 +1,28 @@
+
+operations_count = 0
+
+def main():
+    ask_again = True
+    while(ask_again):
+        a = input("Enter the numerator: ")
+        b = input("Enter the denominator: ")
+        result = perform_division(a,b)
+        print(result)
+        ask_again = input("Do you want to perform another operation? Enter yes or no: ")
+        if(ask_again == 'yes'):
+            ask_again = True
+        else:
+            ask_again = False
+            print("You performed " + str(operations_count) + " operations, bye!")

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are concatenating a string inside a loop. Concatenating immutable sequences will always results in a new object. This means that building up a sequence by repeated concatenation will have a quadratic runtime cost in the total sequence length. For efficiency and to get a linear runtime cost, consider appending values to a list and concatenating that list using [str.join()](https://docs.python.org/3/library/stdtypes.html#str.join) or [bytes.join()](https://docs.python.org/3/library/stdtypes.html#bytes.join) method.

[Learn more](https://docs.python.org/3/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together)